app/main.py

```diff
@@ -1,3 +1,58 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        elif isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Distance | int | float) -> bool:
+        other_distance = other.km \
+            if isinstance(other, Distance) else other
+        return self.km < other_distance
+
+    def __gt__(self, other: Distance | int | float) -> bool:
+        other_distance = other.km \
+            if isinstance(other, Distance) else other
+        return self.km > other_distance
+
+    def __eq__(self, other: Distance | int | float) -> bool:
+        other_distance = other.km \
+            if isinstance(other, Distance) else other
+        return self.km == other_distance
+
+    def __le__(self, other: Distance | int | float) -> bool:
+        other_distance = other.km \
+            if isinstance(other, Distance) else other
+        return self.km <= other_distance
+
+    def __ge__(self, other: Distance | int | float) -> bool:
+        other_distance = other.km \
+            if isinstance(other, Distance) else other
+        return self.km >= other_distance
```

Comment:
You could think over the implementation of some methods basing them on the logic of others. E.g. `self` is greater or equal to `other` when `self` is not less than `other`.